pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -982,9 +983,6 @@ func keyfileWatcher(ctx context.Context, watcher *fswatcher.Watcher, keyfilePath
 			// with ipsec.UpsertIPsecEndpoint()
 			nodeHandler.AllNodeValidateImplementation()
 
-			// Publish the updated node information to k8s/KVStore
-			nodediscovery.UpdateLocalNode()
```

Comment:
I'm not sure I understand. Is this not required anymore because this push to k8s or kvstore now happens as part of the `nodeUpdate` called via `AllNodeValidateImplementation` above? Is that what "the LocalNodeStore [...] to
trigger a new synchronization whenever appropriate" refers to?

Reply:
Kinda. It is actually the `node.SetIPsecKeyIdentity` call that eventually causes the update of the CiliumNode resource and/or kvstore representation, because it triggers an update of the LocalNodeStore. Which, in turn, emits an [event](https://github.com/giorio94/cilium/blob/mio%2Flocal-node-refactor/pkg/node/local_node_store.go#L158), that is asynchronously received by the [nodediscovery logic](https://github.com/giorio94/cilium/blob/mio%2Flocal-node-refactor/pkg/nodediscovery/nodediscovery.go#L228-L230), which performs the actual update.

Reply:
I see. Thanks for the pointers!